### ISSUE TITLE
Doesn't build with cabal 1.16 with ghc 7.6.3 

### ISSUE THREAD

**Issue by dilawar** [ku-fpg/sized-types#4]
Fails with following error:

~~~~~~~~
Data/Sized/Sparse/Matrix.hs:61:26:
    Could not deduce (Eq a) arising from a use of `prune'
    from the context (Size m, Size n, Size m', Size n', n ~ m', Num a)
      bound by the type signature for
                 mm :: (Size m, Size n, Size m', Size n', n ~ m', Num a) =>
                       Matrix (m, n) a -> Matrix (m', n') a -> Matrix (m, n') a
      at Data/Sized/Sparse/Matrix.hs:50:7-112
    Possible fix:
      add (Eq a) to the context of
        the type signature for
          mm :: (Size m, Size n, Size m', Size n', n ~ m', Num a) =>
                Matrix (m, n) a -> Matrix (m', n') a -> Matrix (m, n') a
    In the expression: prune 0 s2
    In a pattern binding: (Matrix _ mp2) = prune 0 s2
    In an equation for `mm':
        mm s1 s2
          = Matrix 0 mp
          where
              mp
                = Map.fromList
                    [((x, y), v) |
                       (x, y) <- M.all, let ..., not (Set.null s), let ..., v /= 0]
              (Matrix _ mp1) = prune 0 s1
              (Matrix _ mp2) = prune 0 s2
              rs = rowSets (Map.keysSet mp1)
              ....
Failed to install sized-types-0.3.4.0
cabal: Error: some packages failed to install:
~~~~~~

**Comment by xich** [comment]
Hmmm, 0.3.4 is an old version. This repo has 0.5.0 in it, which has `Eq` constraints as you expect. Are you cabal installing from hackage? Maybe try to figure out why it's not choosing 0.5.0?

**Comment by RyanGlScott** [comment]
One possibility is that `cabal` is falling back on an old version due to `sized-types-0.5.0` requiring `singletons == 0.10.*`, whereas the latest version is 1.1.

**Comment by xich** [comment]
Ah, maybe someone can see if it works with the singletons and report back? I can always bump the bounds and release a new sized-types... I'm just not familiar with this package (haven't used it in years) so not sure how to test it myself.

**Comment by RyanGlScott** [comment]
Sure, I can try it out. I'll update it for GHC 7.10 as well while I'm at it.

**Comment by RyanGlScott** [comment]
I just tested it with `singletons-1.1`, and everything works as expected (all the tests seem to pass). I bumped the version number to 0.5.1 as well.

**Comment by RyanGlScott** [comment]
I'm glad I enabled Travis-CI integration, because I also discovered a hidden host of GHC 7.6-specific errors. They should all be fixed now, however, so we should release a new version of `sized-types` to Hackage.

**Comment by RyanGlScott** [comment]
Released to Hackage as [`sized-types-0.5.1`](http://hackage.haskell.org/package/sized-types).
